### ISSUE TITLE
Security and Setup Check Fails to Include NextCloud Directory in PROPFIND Request

### ISSUE THREAD

**Issue by bj-github** [nextcloud/server#46386]
Running: Nextcloud Hub 8 (29.0.3)

I looked at the Apache logs after opening the Administration Overview page, and getting the numerous warnings oft-repeated elsewhere, and found in the `access_log` that **the initial `PROPFIND` lacked the `<dir>` subdirectory in which my NextCloud instance is installed**:

```
... "PROPFIND /remote.php/webdav HTTP/1.1" 404 10086 "-" "Nextcloud Server Crawler"
... "HEAD /<dir>//home/<user>/ncdata/.ocdata HTTP/1.1" 404 - "-" "Nextcloud Server Crawler"
... "HEAD /<dir>/apps/settings/js/map-test.js.map HTTP/1.1" 404 - "-" "Nextcloud Server Crawler"
... "HEAD /<dir>/apps/settings/js/esm-test.mjs HTTP/1.1" 404 - "-" "Nextcloud Server Crawler"
... "HEAD /<dir>/ocm-provider/ HTTP/1.1" 404 - "-" "Nextcloud Server Crawler"
... "HEAD /<dir>/ocs-provider/ HTTP/1.1" 404 - "-" "Nextcloud Server Crawler"
... "GET /.well-known/webfinger HTTP/1.1" 404 10094 "-" "Nextcloud Server Crawler"
... "HEAD /<dir>/core/fonts/NotoSans-Regular-latin.woff2 HTTP/1.1" 404 - "-" "Nextcloud Server Crawler"
```

Also, fwiw, is the following in Apache's `error_log` presumably because, without the `/<dir>` prefix in the `PROPFIND` request, there is no `remote.php` script to be found?:

```
[proxy_fcgi:error] [pid ...] [client <serverIP#>:54070] AH01071: Got error 'Primary script unknown'
```

_Originally posted by @bj-github in https://github.com/nextcloud/server/issues/44114#issuecomment-2216272626_

**Comment by joshtrichards** [comment]
No idea since you didn't fill out the bug reporting template. 

Wild guess... check your `overwritewebroot` config value: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#overwritewebroot

**Comment by bj-github** [comment]
Sorry about that. Choosing the "Reference in new issue" option off a comment didn't take me anywhere near the workflow I now see I get if I select `[New Issue]` directly.

My `config.php` has each of `htaccess.RewriteBase` and `overwrite.cli.url` specifying the subdirectory appropriately. I have no `overwritewebroot` setting.

Following the clue of someone else's comment, I checked my `/etc/hosts` file. And, sure enough, the wrong IPv6 address was mapped to the Domain under which NextCloud is accessed. I don't recall ever having a need to look at that file in the decade+ I've had the device up and running. So I can only guess this started happening in recent times (since around Hub 7, as I recall) because something on the machine changed that triggered the Administration Overview to start using IPv6 - via that erroneous `/etc/hosts` entry - instead of IPv4. Perhaps it was as simple as a (non-NextCloud, obviously!) system update that added that erroneous line after the valid IPv4 address line?

(I had consoled myself in the meantime with `scan.nextcloud.com` reporting that things were ok. Thank you for that service.)

Anyway, with that fixed, all the issues reported under "Security & setup warnings" have disappeared. So thank you for your patience. And perhaps it'll leave a little bit of a "breadcrumb" to help someone else encountering the same issue.
